sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventData.cs

```diff
@@ -13,22 +14,25 @@ public abstract class AuthenticationEventData
         /// <summary>Gets the event identifier.</summary>
         /// <value>The event identifier.</value>
         [JsonPropertyName("tenantId")]
-        public Guid TenantId { get; set; }
+        [Required]
+        public Guid? TenantId { get; set; } = null;
```

Comment:
If it's not `Guid?`, then required doesn't quite work as expected as `Activator.CreateInstance` can set the GUID to default value of Guid.Empty.

Reply:
Have created a new validator for this called RequiredNonDefault. Will Add that for now.